### PR TITLE
fix: Date argument non-optional for the onChange prop

### PR DESCRIPTION
**Pull request by sakibian**
Closes: #1601

### CODE REVIEW

**Comment by sakibian** [comment]
I have read the CLA Document and I hereby sign the CLA

File: packages/components/src/DateInput.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ const DATE_FORMAT = 'YYYY-MM-DD';
 
 type DateInputProps = {
   className?: string;
-  onChange?: (date?: string) => void;
+  onChange: (date: string) => void; // Remove the question mark to make date non-optional
```
Comment (mofojed) [inline packages/components/src/DateInput.tsx:15]: No need to add this comment here. 
```suggestion
  onChange: (date: string) => void;
```

File: packages/components/src/DateInput.tsx [hunk 2]
```diff
@@ -23,7 +23,7 @@ const DateInput = React.forwardRef<HTMLInputElement, DateInputProps>(
   (props: DateInputProps, ref) => {
     const {
       className = '',
-      onChange = () => undefined,
+      onChange = () => undefined, // Provide a default value for onChange
```
Comment (mofojed) [inline packages/components/src/DateInput.tsx:26]:
```suggestion
      onChange = () => undefined,
```

File: packages/components/src/DateInput.tsx [hunk 3]
```diff
@@ -62,16 +62,6 @@ const DateInput = React.forwardRef<HTMLInputElement, DateInputProps>(
     );
   }
 );
-
 DateInput.displayName = 'DateInput';
 
-DateInput.defaultProps = {
```
Comment (mofojed) [inline packages/components/src/DateInput.tsx:68]: Good cleanup!

**Review by mofojed — CHANGES_REQUESTED** [review]
Just remove the unnecessary comments. Thank you for the contribution!

**Review by mofojed — DISMISSED** [review]
Thank you for the contribution!

File: packages/code-studio/src/styleguide/Inputs.tsx [hunk 4]
```diff
@@ -85,6 +85,10 @@ function Inputs(): React.ReactElement {
     []
   );
 
+  const handleChange = (date: string) => {
```
Comment (mofojed) [inline packages/code-studio/src/styleguide/Inputs.tsx:88]: We've still got linter issues here - our formatting is fairly strict, we include some extensions and VS Code settings for automatically formatting code on save: https://github.com/deephaven/web-client-ui/blob/main/.vscode/settings.json

For this case though, I'm going to recommend the following:
- Make the `onChange` prop on `DateInput` optional still (the ticket was specifically for the `date` parameter on the callback to be marked as required, we don't need to make `onChange` itself required as well)
- Don't add an `onChange` here in the styleguide.

That would make it consistent with our existing behaviour/other components (such as `DateTimeInput`).

**Review by mofojed — CHANGES_REQUESTED** [review]

**Review by mofojed — APPROVED** [review]

**Pull request merged**